storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -203,13 +218,29 @@ void initializeMesosStormConf(Map conf, String localDir) {
 
     _allowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_ALLOWED_HOSTS));
     _disallowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_DISALLOWED_HOSTS));
+
+    Set<String> zooKeeperServers = listIntoSet((List<String>) conf.get(Config.STORM_ZOOKEEPER_SERVERS));
+    String zooKeeperPort = String.valueOf(conf.get(Config.STORM_ZOOKEEPER_PORT));
+    _logviewerZkDir = Optional.fromNullable((String) conf.get(Config.STORM_ZOOKEEPER_ROOT)).or("") + "/storm-mesos/logviewers";
+    LOG.info("Logviewer information will be stored under {}", _logviewerZkDir);
+
+    if (zooKeeperPort == null || zooKeeperServers == null) {
+      LOG.error("ZooKeeper configs are not found in storm.yaml");
+    } else {
+      StringBuilder connectionString = new StringBuilder();
+      for (String server : zooKeeperServers) {
+        connectionString.append(String.format("%s:%s,", server, zooKeeperPort));
+      }
+      _zkClient = new ZKClient(connectionString.substring(0, connectionString.length() - 1));
```

Comment:
This might be better as: `_zkClient = new ZKClient(connectionString.toString());`

Reply:
@JessicaLHartog I used substring( ) because I was scared the extra appended ',' at the end of the connection string would screw things up

Reply:
We've changed this logic around a bit in our latest "minor cleanup commit".

Reply:
Here's the commit where we tweaked the logic:
* https://github.com/mesos/storm/commit/72681481f4df0ab33da08f7b6e9d9ba3db17c85e